nix/ext/scripts/update_versions_json.py

```
@@ -0,0 +1,104 @@
+                    peeled_tags[tag] = commit_hash
+                else:
+                    # Regular tag entry
+                    tag = ref.removeprefix("refs/tags/")
+                    try:
+                        parse_version(tag)
+                    except InvalidVersion:
+                        continue
+                    tags[tag] = commit_hash
+
+    # Second pass: prefer peeled commit hashes when available
+    for tag, commit_hash in tags.items():
```

Ruff (0.14.13)

Loop control variable `commit_hash` not used within loop body

(B007)